Sources/WalletConnectRelay/Dispatching.swift

```diff
@@ -118,8 +126,19 @@ final class Dispatcher: NSObject, Dispatching {
         socket.onDisconnect = { [unowned self] error in
             self.socketConnectionStatusPublisherSubject.send(.disconnected)
             if error != nil {
-                self.socket.request.url = relayUrlFactory.create()
+                self.socket.request.url = relayUrlFactory.create(fallback: fallback)
+            }
+            Task(priority: .high) {
+                await self.socketConnectionHandler.handleDisconnection()
             }
+        }
+    }
+    
+    private func handleFallbackIfNeeded(error: NetworkError) {
+        if error == .webSocketNotConnected && socket.request.url?.host == NetworkConstants.defaultUrl {
```

Comment:
can you help me understand the logic? it will fallback if socket did not connected and url is `relay.walletconnect.com`? 
could the fallback also happen when socket did not connected from different reasons?